catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java

```diff
@@ -199,6 +199,11 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
 		public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
 			if (Utils.checkIsNetworkAvailableAndShowErrorMessage(WebViewActivity.this)) {
 				ToastUtil.showError(getBaseContext(), R.string.error_unknown_error);
+			} else {
```

Comment:
actually the old way by showing the toast was completely wrong, cause the error roast will be shown even if the error isn't a network error. for more info please have a look at errors that this callback handle. [webViewErrors](https://developer.android.com/reference/kotlin/android/webkit/WebViewClient#constants)

to indicate no internet access, just compare `errorCode`  or `description` with the corresponding values, which means no need to check if the device has internet access in the first place.